internal/pkg/status/status.go

```diff
@@ -14,41 +14,175 @@ limitations under the License.
 package status
 
 import (
+	"context"
 	"fmt"
 	"io"
 
 	"gopkg.in/src-d/go-git.v4/plumbing/object"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes"
+	//metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cli-experimental/internal/pkg/client"
+	clientu "sigs.k8s.io/cli-experimental/internal/pkg/client/unstructured"
 	"sigs.k8s.io/cli-experimental/internal/pkg/clik8s"
 )
 
+// Condition types
+const (
+	// Level Conditions
+
+	// ConditionReady - Indicates the object is resource for use
+	ConditionReady ConditionType = "Ready"
+	// ConditionSettled - Indicates the controller is done reconciling the spec
+	// This is not implemented yet
+	ConditionSettled ConditionType = "Settled"
+
+	// Terminal condition
+
+	// ConditionFailed - The resource is in failed condition and the controller will not process it further
+	ConditionFailed ConditionType = "Failed"
+	// ConditionCompleted - The resource is done doing what it intends. Example Job, Pods can have completed state.
+	ConditionCompleted ConditionType = "Completed"
+	// Terminating - Indicates the resource is being deleted.
+	ConditionTermination ConditionType = "Terminating"
+
+	// Progress condition
+
+	// ConditionProgress - Indicates the controller is still working to satisfy the intent in the resource spec.
+	ConditionProgress ConditionType = "Progress"
+)
+
 // Status returns the status for rollouts
 type Status struct {
+	// DynamicClient is the client used to talk
+	// with the cluster
+	DynamicClient client.Client
+	// Out stores the output
+	Out io.Writer
+	// Resources is a list of resource configurations
 	Resources clik8s.ResourceConfigs
-	Out       io.Writer
-	Clientset *kubernetes.Clientset
-	Commit    *object.Commit
+	// Commit is a git commit object
+	Commit *object.Commit
```

Comment:
is this still used?

Reply:
Iam retaining this for future use. Will sync up with @pwittrock on this.